File: arrays-and-strings/src/max_consecutive_ones_n_linear_solution.java
public class max_consecutive_ones_n_linear_solution{
    public int findMaxConsecutiveOnes(int[] nums) {

        int max_l = Integer.MIN_VALUE;
        int i = 0;

        while(i < nums.length){
            if(nums[i] == 1){
                int curr_length = 1;
                int nextZeroIterator = i + 1;
                while(nextZeroIterator < nums.length && nums[nextZeroIterator] == 1){
                    nextZeroIterator++;
                    curr_length++;
                }
                i = nextZeroIterator;
                if(curr_length > max_l) max_l = curr_length;
            } else {
                i++;
            }
        }
        return max_l;
    }
}

File: arrays-and-strings/src/max_consecutive_ones_n_linear_solution.cpp
#include <algorithm>
#include <vector>
#include <iostream>

using namespace std;

class Solution
{
    public:
        int findMaxConsecutiveOnes(vector<int> &nums)
        {
            int max_l = 0;

            int i = 0;

            while(i < nums.size())
            {
                if(nums[i] == 1)
                {
                    int curr_len = 1;
                    int r = i + 1;
                    while(r < nums.size() && nums[r] == 1)
                    {
                        curr_len++;
                        r++;
                    }
                    i = r;
                    max_l = max(max_l, curr_len);
                }
                else
                {
                    i++;
                }
            }

            return max_l;
        }
};

int main()
{
    vector<int> nums = {1,1,0,1,1,1};

    Solution sol;

    int p = sol.findMaxConsecutiveOnes(nums);

    cout << p << endl;

    return 0;
}